great_expectations/checkpoint/actions.py

```diff
@@ -103,8 +108,79 @@ def filter_results(self, class_: Type[ValidationAction]) -> list[dict]:
         return [action_result for action, action_result in self._data if isinstance(action, class_)]
 
 
+class ValidationActionRegistry:
+    """
+    Registers ValidationActions to enable deserialization based on their configuration.
+
+    Uses the `type` key from the action configuration to determine which registered class
+    to instantiate.
+    """
+
+    def __init__(self):
+        self._registered_actions: dict[str, Type[ValidationAction]] = {}
+
+    def register(self, action_type: str, action_class: Type[ValidationAction]) -> None:
+        """
+        Register a ValidationAction class with the registry.
+
+        Args:
+            action_type: The type of the action to register.
+            action_class: The ValidationAction class to register.
+
+        Raises:
+            ValidationActionAlreadyRegisteredError: If the action type is already registered.
+        """
+        if action_type in self._registered_actions:
+            raise ValidationActionAlreadyRegisteredError(action_type)
+
+        self._registered_actions[action_type] = action_class
+
+    def get(self, action_type: str | None) -> Type[ValidationAction]:
```

Comment:
Doesn't this not work for 3.9?

Reply:
`str | None` vs `Optional[str]`? I think that's fine?

Reply:
I guess you're right! I have no idea what I'm doing!